src/redash_stmo/datasource_link/bundle/datasource_link.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+
+class DatasourceLink extends React.Component {
+  static propTypes = {
+    clientConfig: PropTypes.object.isRequired,
+    datasourceId: PropTypes.number.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      type_name: '',
+      doc_url: '',
+    };
+  }
+
+  _loadURLData() {
+    fetch(`${this.props.clientConfig.basePath}api/data_sources/${this.props.datasourceId}/link`)
+      .then((response) => {
+        if (response.status === 200) {
+          return response.json();
+        }
+        return {};
+      })
+      .then((json) => {
+        const { type_name, doc_url } = json.message;
```

Comment:
Forgive my noobness with modern JS, wouldn't this code fail in case the fetch call returns an empty object because for some error happening server side?

I mean, `json.message` would not exist and then the variable unpacking would fail. Does this require more defensive code?

Reply:
Good catch. I will fix this prior to merging, thanks!